### PR TITLE
GitHub content

### PR DESCRIPTION
**Pull request by ktyle**
Initial stab at organizing content for GitHub, addressing #99 . I revamped the initial "getting-started-with-github" page and created a github subdirectory for what is currently 10 individual lessons (all of which are in very provisional rough-draft status).

### CODE REVIEW

**Comment by ktyle** [comment]
Thanks @brian-rose and @clyne ... I will continue work on this and incorporate your suggestions.

**Comment by clyne** [comment]
How can I help, @ktyle ? I'm sure I can cobble something up for one of the less advanced topics that could be refined my more knowledgeable folks

**Comment by SimonGoring** [comment]
One suggestion I'd like to make is that the way I've framed it in the past is by introducing/motivating the need for version control first (and introducing `git` as a concept) and then talking about GitHub specifically (see [my slides here](https://github.com/throughput-ec/ThesisIsCode/blob/main/git/03_gitSlides.md)).  This way you don't lock people into the platform specifically.

I've run into folks who only know how to use GitHub Desktop for their version control, and then miss out on integrated GUIs like GitKraken or other tools that let them work with other VC platforms.

I suppose my suggestion, in simpler terms, is to introduce `git` as the *de facto* standard, and then GitHub as a platform for collaboration around `git`.

**Comment by dopplershift** [comment]
@SimonGoring I can definitely appreciate the idea that `git` is the core tool, and GitHub is one of many online sites built around `git`. My memory of our discussions was that we really wanted to present `GitHub` (as the core tool we use for Pythia) as an important site for collaboration and interaction with open source (opening issues, having discussions, etc.). It's possible to use these important collaborative functions without even knowing (or being sold on) version control and `git`.

**Comment by brian-rose** [comment]
This is a good discussion, and I think there are pros and cons to leading with version control / git before talking about GitHub. But yes as @dopplershift says, we want to aim for a slightly different narrative that emphasizes "GitHub as a collaboration platform" first. I haven't seen many examples of tutorials like that, and it could be one bit of value that Pythia can add. 

If the open science community (and maybe Pythia along with it) migrates away from GitHub to another platform, then we'll have to revisit. But of course that's true of Python too!

**Comment by ktyle** [comment]
I've incorporated suggestions above into another draft outline, which can be viewed [here](http://www.atmos.albany.edu/facstaff/ktyle/pythia/foundations/_build/html/foundations/getting-started-github.html ). We will discuss more at today's EWG call!

**Comment by brian-rose** [comment]
Same build failure as #174

**Review by clyne — APPROVED** [review]

**Review by brian-rose — APPROVED** [review]
@ktyle great work, this looks like a really compelling outline.

Just one comment (and this should NOT hold up merging this PR): I still like the idea we discussed at earlier meetings of some kind of "safe sandbox" for practicing opening pull requests, i.e. some bare bones repo with something in plain text. 

We could add a bullet point on the "Opening a Pull Request on GitHub" page about creating a resource like this to be used as part of the lessons in that subsection. This could create a nice segue into the next section, "Contribute to Project Pythia via GitHub" -- where the user "graduates" from the safe sandbox to a real "production" repo.

**Comment by dopplershift** [comment]
So just a point: You can always make your own repository and make pull requests against it. Are there good benefits to  running such a sandbox for our learners that justify the cost of managing it? The only differences I can think of are:
* Getting human feedback (this depends on having people dedicated to watching for PRs)
* Dealing with multiple git remotes

**Comment by brian-rose** [comment]
> So just a point @brian-rose about a PR sandbox: You can always make your own repository and make pull requests against it. Are there good benefits to running such a sandbox for our learners that justify the cost of managing it? The only differences I can think of are:
> 
> * Getting human feedback (this depends on having people dedicated to watching for PRs)
> * Dealing with multiple git remotes

Good question, @dopplershift. I was *not* thinking that users would get human feedback on these "practice" PRs. 

I suppose I'm thinking vaguely of the "make your own repository" step as a barrier that we can reduce by providing a bare-bones repo along with step-by-step instructions on how to fork it, create a branch, commit a change, and open a PR. We could certainly do the same with one of our "production" repos, but I think there may be a benefit to building a tutorial around a *simple* repo with fewer files and less complication around things like linters etc.

**Comment by ktyle** [comment]
Thanks @clyne and @brian-rose ; now that the jupyterbook build is working again, I will proceed with further edits to these relevant pages.

**Comment by dopplershift** [comment]
@brian-rose I guess with a repo running the [stale bot](https://github.com/marketplace/stale) we wouldn't end up with a repo full of random PRs never to be dealt with.

We *definitely* do NOT want to have instructions leading people to make test/learning PRs against our productions repos. I saw what happened to the poor numpy devs when some random online course had instructions that led to people doing just that. It was terrible.

**Comment by ktyle** [comment]
The (currently) 11 GitHub chapter sections now follow the Pythia template. 

Next up: add the actual content to these sections.

**Comment by brian-rose** [comment]
> The (currently) 11 GitHub chapter sections now follow the Pythia template.
> 
> Next up: add the actual content to these sections.

This is really nicely polished now @ktyle! Is it time to click the "Ready for review" button and get this merged?

**Comment by ktyle** [comment]
@brian-rose ready for review!

**Review by brian-rose — APPROVED** [review]

**Comment by brian-rose** [comment]
@clyne, @kmpaul, just looking for one more final approval before merging this.

**Comment by clyne** [comment]
> @clyne, @kmpaul, just looking for one more final approval before merging this.

Sorry. I thought I'd already approved this. Done.

**Review by clyne — APPROVED** [review]
Looks great!

**Comment by brian-rose** [comment]
> > @clyne, @kmpaul, just looking for one more final approval before merging this.
> 
> Sorry. I thought I'd already approved this. Done.

No worries, there were just some revisions after the last round of comments.

I think we'll call this done! Pushing the green button and giving thanks...
